plugins/window_size/lib/src/window_size_channel.dart

```diff
@@ -26,31 +26,60 @@ const String _windowSizeChannelName = 'flutter/windowsize';
 ///
 /// Returns a list of screen info maps; see keys below.
 const String _getScreenListMethod = 'getScreenList';
+
 /// The method name to request information about the window containing the
 /// Flutter instance.
 ///
 /// Returns a list of window info maps; see keys below.
 const String _getWindowInfoMethod = 'getWindowInfo';
+
 /// The method name to set the frame of a window.
 ///
 /// Takes a frame array, as documented for the value of _frameKey.
 const String _setWindowFrameMethod = 'setWindowFrame';
 
+/// The method name to set the minimum size of a window.
+///
+/// Takes a window size array, as documented for the value of _windowSizeKey.
+const String _setWindowMinimumSizeMethod = 'setWindowMinimumSize';
+
+/// The method name to set the maximum size of a window.
+///
+/// Takes a window size array, as documented for the value of _windowSizeKey.
+const String _setWindowMaximumSizeMethod = 'setWindowMaximumSize';
+
+/// The method name to get the minimum size of a window.
+///
+/// Returns a window size array, as documented for the value of _windowSizeKey.
+const String _getWindowMinimumSizeMethod = 'getWindowMinimumSize';
+
+/// The method name to get the maximum size of a window.
+///
+/// Returns a window size array, as documented for the value of _windowSizeKey.
+const String _getWindowMaximumSizeMethod = 'getWindowMaximumSize';
```

Comment:
I think the method channel needs to clearly define what "no constraint" returns, and how to set it. Currently you're leaking whatever the platform implemention is (e.g., `FLT_MAX` for macOS) which would make removing constraints, or checking if there is a constraint, difficult on the Dart side, requiring platform-specific switching.

Something should be defined for the channel (null argument maybe?) and the platform implementions should translate to the right representation is for the platform.

Reply:
Null feels kinda ick, especially in the days of not null by design. I can't think of anything better, apart from using a string and having a known value for unconstrained.

Reply:
It doesn't need to be a string; we could define `0` to be unconstrained. For min size, that's equivalent to unconstrained anyway, and for max 0 doesn't make sense, so would never come up in normal usage. Similarly, we could use `-1` since we're not dealing with unsigned ints, and `-1` would never be used.

Reply:
One upside to using `FLT_MAX` as unconstrained sizing for max width and max height is that `Size.fromWidth` and `Size.fromHeight` work out of the box. It looks like Flutter is already using float max as sentinel for unconstrained maximum size.

Reply:
But it relies on every native implementation having the same definition of float size; there's no real reason that needs to be consistent across all languages. `FLT_MAX` doesn't even have a single defined value in C++ for instance.

I would much rather we pick something clear and explicit, and convert in every implementation, than have something that happens to work right now but isn't guaranteed to have consistent behavior in the future.

Reply:
So, how about `double.infinity` in the exposed Dart API for congruence with `Size.fromHeight` and `Size.fromWidth`, and `-1` for across the channel from Dart land to the native code? I like `-1` as it is obviously invalid for all of the sizing properties, and I'm hoping `-1f` has a single defined value in C++.

Reply:
> So, how about `double.infinity` in the exposed Dart API for congruence with `Size.fromHeight` and `Size.fromWidth`, and `-1` for across the channel from Dart land to the native code?

Sounds good to me.

> I'm hoping `-1f` has a single defined value in C++.

I'm not sure what you mean exactly; -1 *is* a value. The problem with `FLT_MAX` is that it isn't a value, it’s a concept whose value depends on the size of `float`, which it itself not fixed.

Reply:
I've made an initial pass at always using -1 to mark min or max as unconstrained through the channel. It's been long enough since I've done Obj C that I'm questioning my approach with converting the values, and I'd appreciate guidance.

I also ran into some issues with some of the plugins used in the testbed gaining endorsed macos versions, so I converted everything across. My temporary solution is going to break windows and linux, so I'd appreciate guidance here as well.

/cc @franciscojma86

Reply:
The solution to the endorsed plugin issues you're hitting are that we need to remove all the macOS implementation of Flutter plugins (in another PR). We were waiting for all of them to land, which I think has happened now. I can probably do it tomorrow.

Reply:
https://github.com/google/flutter-desktop-embedding/pull/621

Reply:
Looking at the code, it seems like using `0` for unconstrained minimum size would be much simpler, since it's an obviously meaningful value for lower bounds, and doesn't have any value ambiguity like the max values do. We can still use -1 for unconstrained max, but that eliminates a bunch of conversion.

Reply:
Now using 0 for unconstrained minimum size.

Reply:
The special values for min and max should be documented in these comments.

Reply:
Done